tests/prefetch_related/tests.py

```diff
@@ -1241,3 +1241,41 @@ def test_bug(self):
         prefetcher = get_prefetcher(self.rooms[0], 'house')[0]
         queryset = prefetcher.get_prefetch_queryset(list(Room.objects.all()))[0]
         self.assertNotIn(' JOIN ', force_text(queryset.query))
+
+
+class Ticket25546Tests(TestCase):
+
+    def setUp(self):
+        self.book1, self.book2 = [
+            Book.objects.create(title='book1'),
+            Book.objects.create(title='book2'),
+        ]
+        self.author11, self.author12, self.author21 = [
+            Author.objects.create(first_book=self.book1, name='Author11'),
+            Author.objects.create(first_book=self.book1, name='Author12'),
+            Author.objects.create(first_book=self.book2, name='Author21'),
+        ]
+        self.author1_address1, self.author1_address2, self.author2_address1 = [
+            AuthorAddress.objects.create(author=self.author11, address="Happy place"),
+            AuthorAddress.objects.create(author=self.author12, address="Haunted house"),
+            AuthorAddress.objects.create(author=self.author21, address="Happy place"),
+        ]
+
+    def test_prefetch(self):
+        with self.assertNumQueries(3):
+            books = Book.objects.filter(
+                title__in=['book1', 'book2']
+            ).prefetch_related(
+                Prefetch(
+                    'first_time_authors',
+                    Author.objects.prefetch_related(
+                        Prefetch(
+                            'addresses',
+                            AuthorAddress.objects.filter(
+                                address="Happy place",
+                            ),
+                        )
+                    ),
+                ),
+            )
+            book1, book2 = list(books)
```

Comment:
Assert that accessing prefetched relationships doesn't issue queries (`self.assertNumQueries(0)`).